pkg/config/files.go

```diff
@@ -0,0 +1,126 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/mitchellh/go-homedir"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	DefaultUserConfigFile   = "~/.microshift/config.yaml"
+	defaultUserDataDir      = "~/.microshift/data"
+	DefaultGlobalConfigFile = "/etc/microshift/config.yaml"
+	defaultGlobalDataDir    = "/var/lib/microshift"
+	// for files managed via management system in /etc, i.e. user applications
+	defaultManifestDirEtc = "/etc/microshift/manifests"
+	// for files embedded in ostree. i.e. cni/other component customizations
+	defaultManifestDirLib = "/usr/lib/microshift/manifests"
+)
+
+var (
+	configFile   = findConfigFile()
+	dataDir      = findDataDir()
+	manifestsDir = findManifestsDir()
+)
+
+func GetConfigFile() string {
+	return configFile
+}
+
+func GetDataDir() string {
+	return dataDir
+}
+
+func GetManifestsDir() []string {
+	return manifestsDir
+}
+
+// Returns the default user config file if that exists, else the default global
+// config file, else the empty string.
+func findConfigFile() string {
+	userConfigFile, _ := homedir.Expand(DefaultUserConfigFile)
```

Comment:
It just occurred to me that we wanted to remove user dir (microshift needs to be run as root anyway), but given the size of the PR and effort spent on reviewing it we might do it in another PR :)